### PR TITLE
Removed addon operator managed manifests and added installation of addon operator

### PR DESCRIPTION
**Pull request by thorfour**
Removed assets that are manged by the kubernetes-addon-operator (KAO)

KAO now installs these assets

### CODE REVIEW

File: modules/tectonic/resources/manifests/operators/addon.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+apiVersion: extensions/v1beta1
```
Comment (yifan-gu) [inline modules/tectonic/resources/manifests/operators/addon.yaml]: nit: maybe rename the file to kubernetes-addon-operator.yaml?

File: modules/tectonic/resources/manifests/operators/appversion/addon.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+apiVersion: tco.coreos.com/v1
```
Comment (yifan-gu) [inline modules/tectonic/resources/manifests/operators/appversion/addon.yaml]: nit: maybe rename to appversion-kubernetes-addon.yaml to be consistent to other files?

**Comment by yifan-gu** [comment]
I think those manifest files should be in `modules/tectonic/resources/manifests/updater/operators`, and `modules/tectonic/resources/manifests/updater/app_versions`

File: modules/tectonic/resources/tectonic.sh [hunk 3]
```diff
@@ -196,6 +194,21 @@ kubectl create -f updater/app-version-kind.yaml
...
+    fi
+done
+
+# Create all operators 
+for manifest in operators/*; do
+    if [ -f $manifest ] 
+    then
+        kubectl create -f manifest
+    fi
+done
```
Comment (yifan-gu) [inline modules/tectonic/resources/tectonic.sh]: We can just add `kubectl create -f update/operators/kubernetes-addon-operator.yaml` below line 219

Comment (yifan-gu) [reply modules/tectonic/resources/tectonic.sh]: Same for the appversion yaml

**Comment by yifan-gu** [comment]
Also need to modify the assets.tf in the `module/upload-payload` folder.
See https://github.com/coreos/tectonic-installer/pull/2375/files#diff-dcf823fbd41185c05532da44b74b888b

So that the payload contains the new operator and its appversion

File: modules/tectonic/resources/tectonic.sh (hunk 3 above)
Comment (thorfour) [reply modules/tectonic/resources/tectonic.sh]: ah cool, didn't know kubectl did that.

**Comment by diegs** [comment]
Please update the PR title and description to reflect the full scope of these changes.

I just kicked off the smoke tests for AWS as well.

File: modules/tectonic/resources/manifests/updater/app_versions/app-version-kubernetes-addon.yaml [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+metadata:
+  name: kubernetes-addon
+  namespace: tectonic-system
+  labels:
+    managed-by-channel-operator: "true"
+spec:
+  desiredVersion: ${kubernetes_addon_operator_version}
+  paused: false
+status:
+  currentVersion: ${kubernetes_addon_operator_version}
```
Comment (yifan-gu) [inline modules/tectonic/resources/manifests/updater/app_versions/app-version-kubernetes-addon.yaml]: for the install mode, shouldn't we leave the currentVersion empty?

Comment (thorfour) [reply modules/tectonic/resources/manifests/updater/app_versions/app-version-kubernetes-addon.yaml]: yup fixing

**Comment by cpanato** [comment]
@thorfour @diegs Guys, if you change the manifests you need to run the tests in all platforms. Azure is failing because the `kube-dns`was removed. Bootkube wait for this pod comes up.

@sym3tri

**Comment by sym3tri** [comment]
@thorfour @diegs @yifan-gu @Quentin-M 

Yes... please run tests for all platforms, not just AWS.

**Comment by thorfour** [comment]
That's my fault I apologize, I assumed the automated tests covered everything. Is there a github hook to run all platforms or is that a manual only procedure?

**Comment by cpanato** [comment]
@thorfour https://github.com/coreos/tectonic-installer/blob/master/tests/README.md

**Comment by thorfour** [comment]
Didn't RTFM, my bad

**Pull request merged**